Update combine_study_design.R

TylerSagendorf opened the pull request:
Do not modify `samples$ReporterName`. Ensure references in `samples$ReporterAlias` match those in `references$Reference`. Remove `remove_duplicate_digits` function and associated code, since there are no duplicates within either the PASS1A or PASS1C samples. It is fine if the same ReporterAlias is present in both PASS studies.

TylerSagendorf committed:
Update combine_study_design.R

Do not modify ReporterName column, ensure samples$ReporterAlias match references$Reference, remove `remove_duplicate_digits` section, since there are no duplicates within either the PASS1A or PASS1C samples. It is fine if the same ReporterAlias is present in both PASS studies.

biodavidjm reviewed (commented):
Thanks Tyler, it works on my side. But I have a concern:

in the previous version, I added a function to remove duplicates. I am afraid that removing that function is going to be a problem. If two vial labels are identical in **one of the tmt experiments**, i.e., one or many repetitions in the same time point in the same experiment, this should be a problem. Am I wrong?

TylerSagendorf commented:
@biodavidjm Does this issue ever occur? Viallabels should be unique for a given tissue and experiment.

TylerSagendorf commented:
> @biodavidjm Does this issue ever occur? Viallabels should be unique for a given tissue and experiment.

Rather, has this been an issue when processing 1A and 1C data separately?

TylerSagendorf committed:
Update combine_study_design.R

Update handling of duplicate ReporterAlias and MeasurementName. Make handling of references$Reference entries more robust when references can be a combination of multiple channels.

TylerSagendorf committed:
Update combine_study_design.R

Remove stringr package. Explicitly specify keep.source = TRUE in parse (https://stackoverflow.com/a/59931043).

TylerSagendorf committed:
Update combine_study_design.R

Fixed unintended behavior: "1A" was incorrectly being appended to all entries of study_design$references$Reference, so that make_results_ratio failed (create_crosstab) when it could not find the appropriate reference entries for 1C.
